<launch>
  <arg name="manager"             default="pointcloud_camera_manager"/>
  <arg name="tf_prefix"           default=""/>
  <arg name="rosbag_filename"     default=""/>
  <arg name="depth_width"         default="640"/>
  <arg name="depth_height"        default="480"/>
  <arg name="enable_depth"        default="true"/>
  <arg name="depth_fps"           default="30"/>
  <arg name="enable_pointcloud"   default="false"/>
  <arg name="enable_sync"         default="false"/>
  <arg name="align_depth"         default="false"/>

  <arg name="base_frame_id"             default="$(arg tf_prefix)_link"/>
  <arg name="depth_frame_id"            default="$(arg tf_prefix)_depth_frame"/>

  <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" output="screen"/>
  <node pkg="nodelet" type="nodelet" name="pointcloud_camera" args="load pointcloud_camera/PointCloudNodeFactory $(arg manager)">
    <param name="rosbag_filename"          type="str"  value="$(arg rosbag_filename)"/>
    <param name="enable_pointcloud"        type="bool" value="$(arg enable_pointcloud)"/>
    <param name="enable_sync"              type="bool" value="$(arg enable_sync)"/>
    
    <param name="depth_width"              type="int"  value="$(arg depth_width)"/>
    <param name="depth_height"             type="int"  value="$(arg depth_height)"/>
    <param name="enable_depth"             type="bool" value="$(arg enable_depth)"/>

    <param name="depth_fps"                type="int"  value="$(arg depth_fps)"/>
    
    <param name="base_frame_id"            type="str"  value="$(arg base_frame_id)"/>
    <param name="depth_frame_id"           type="str"  value="$(arg depth_frame_id)"/>
  </node>
</launch>
